### PR TITLE
Jl/caip25 permission migration/update mutators

### PR DESCRIPTION
**Pull request by jiexi**
## **Description**

Replaces existing caveat mutators. Handles ensuring `wallet:eip155` is only upserted for permissions granted to snaps.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28709?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by jiexi** [comment]
just tested manually. seems to work as expected

File: ui/components/multichain/network-list-menu/network-list-menu.tsx [hunk 1]
```diff
@@ -297,7 +297,7 @@ export const NetworkListMenu = ({ onClose }: { onClose: () => void }) => {
           }
 
           if (permittedAccountAddresses.length > 0) {
-            addPermittedChain(selectedTabOrigin, network.chainId);
+            dispatch(addPermittedChain(selectedTabOrigin, network.chainId));
```
Comment (jiexi) [inline ui/components/multichain/network-list-menu/network-list-menu.tsx:300]: Fixing this upstream in develop here https://github.com/MetaMask/metamask-extension/pull/28741

**Review by adonesky1 — APPROVED** [review]
LGTM other than figuring out what's going on with the permission auto-granting piece

File: ui/components/multichain/network-list-menu/network-list-menu.tsx (hunk 1 above)
Comment (jiexi) [reply ui/components/multichain/network-list-menu/network-list-menu.tsx:300]: nevermind, this is the result of a bad merge against develop. Develop actually properly dispatches this

Comment (jiexi) [reply ui/components/multichain/network-list-menu/network-list-menu.tsx]: resolved in the CAIP-25 permission branch upstream of this

**Pull request merged**